lib/ansible/modules/cloud/azure/azure_rm_aci.py

```diff
@@ -0,0 +1,478 @@
+                                    None,
+                                    ip_address,
+                                    self.os_type)
+
+        try:
+            client = ContainerInstanceManagementClient(self.azure_credentials, self.subscription_id)
+            response = client.container_groups.create_or_update(self.resource_group, self.name, parameters)
+
+        except CloudError as exc:
+            self.log('Error attempting to create the container instance.')
+            self.fail("Error creating the ACI instance: {0}".format(str(exc)))
+        return create_aci_dict(response)
+
+    def delete_aci(self):
+        '''
+        Deletes the specified container instance in the specified subscription and resource group.
+        The operation does not delete other resources created as part of creating a container service,
+        including storage accounts, VMs, and availability sets.
+        All the other resources created with the container service are part of the same resource group and can be deleted individually.
+
+        :return: True
+        '''
+        self.log("Deleting the ACI instance {0}".format(self.name))
+        try:
+            client = ContainerInstanceManagementClient(self.azure_credentials, self.subscription_id)
+            response = client.container_groups.delete(self.resource_group, self.name)
+        except CloudError as e:
+            self.log('Error attempting to delete the ACI instance.')
+            self.fail("Error deleting the ACI instance: {0}".format(str(e)))
+
+        return True
+
+    def get_aci(self):
+        '''
+        Gets the properties of the specified container service.
+
+        :return: deserialized ACI instance state dictionary
+        '''
+        self.log("Checking if the ACI instance {0} is present".format(self.name))
+        found = False
+        try:
+            client = ContainerInstanceManagementClient(self.azure_credentials, self.subscription_id)
+            response = client.container_groups.get(self.resource_group, self.name)
+            found = True
+            self.log("Response : {0}".format(response))
+            self.log("ACI instance : {0} found".format(response.name))
+        except CloudError as e:
+            self.log('Did not find the ACI instance.')
+        if found is True:
+            return create_aci_dict(response)
+
+        return False
+
+    def check_need_update(self, cg):
+
+        if self.location.lower() != cg["location"].lower():
+            self.log('Location has changed -- need to delete')
+            return True
+        if self.os_type.lower() != cg["os_type"].lower():
+            self.log('OS Type has changed -- need to delete')
+            return True
+
+        # check if container list has changed
+        if len(cg["containers"]) != len(self.containers):
+            self.log('Number of containers has changed -- need to delete')
+            return True
+
+        for i in range(len(self.containers)):
```

Comment:
sort the container list before comparison? ideally, we don't have to update if just the order changes.